paasta_tools/kubernetes_tools.py

```diff
@@ -795,16 +823,21 @@ def get_pod_template_spec(
         docker_volumes = self.get_volumes(
             system_volumes=system_paasta_config.get_volumes()
         )
+        annotations = {"smartstack_registrations": json.dumps(self.get_registrations())}
+        metrics_provider = self.config_dict.get("autoscaling", {}).get(
+            "metrics_provider", ""
+        )
+        if metrics_provider in {"http", "uwsgi"}:
+            annotations["autoscaling"] = metrics_provider  # type: ignore
```

Comment:
Can you please add a proper type annotation to `annotations` var to avoid this `# type: ignore` workaround?